src/models/core/AnswersConfig.ts

```diff
@@ -77,3 +77,8 @@ export interface AnswersConfigWithToken extends BaseAnswersConfig {
  * @public
  */
 export type AnswersConfig = AnswersConfigWithApiKey | AnswersConfigWithToken;
+
+/**
+ * The AnswersConfig after any defaulting has been done from within answers-core.
+ */
+export type AnswersConfigWithDefaulting = AnswersConfig & { endpoints: Required<Endpoints> };
```

Comment:
Could this be private/internal?

Reply:
I think the `@internal` annotations are only necessary for library exports that would otherwise be publicly exported, and this type isn't exported by the library. I was thinking it could be confusing to add it, since it could imply we're exporting it from the library but keeping it hidden, when in reality we're not. On the other hand it looks like we've already marked a few other things as internal, and leaving it unannotated also could be confused between intentionally leaving it unannotated and just forgetting to do so, so I'll add it!